### PR TITLE
instaloader 4.10.1 (new formula)

### PR DESCRIPTION
**Pull request by bevanjkay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/i/instaloader.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Instaloader < Formula
+  include Language::Python::Virtualenv
+
+  desc "Download media from Instagram"
+  homepage "https://instaloader.github.io/"
+  url "https://github.com/instaloader/instaloader/archive/refs/tags/v4.10.1.tar.gz"
+  sha256 "5a2e0519e44bd4005363adc7d4f88ed5967abd9457e119dee180a33c77cc4890"
+  license "MIT"
+
+  depends_on "python@3.11"
```
Comment (chenrui333) [inline Formula/i/instaloader.rb:10]: why not py3.12?

File: Formula/i/instaloader.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class Instaloader < Formula
+  include Language::Python::Virtualenv
+
+  desc "Download media from Instagram"
+  homepage "https://instaloader.github.io/"
+  url "https://github.com/instaloader/instaloader/archive/refs/tags/v4.10.1.tar.gz"
+  sha256 "5a2e0519e44bd4005363adc7d4f88ed5967abd9457e119dee180a33c77cc4890"
```
Comment (chenrui333) [inline Formula/i/instaloader.rb:7]:
```suggestion
  url "https://files.pythonhosted.org/packages/53/a0/49ded81d0134be2e1c22ae4706c35b74594ed1844bd1e9af703fd310d562/instaloader-4.10.1.tar.gz"
  sha256 "902cc8b9569ca7437323199c8e55dbdcd15ea5c8e874c7864e84a36dd55f8584"
```

File: Formula/i/instaloader.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  license "MIT"
+
+  depends_on "python@3.11"
+
+  resource "certifi" do
+    url "https://files.pythonhosted.org/packages/98/98/c2ff18671db109c9f10ed27f5ef610ae05b73bd876664139cf95bd1429aa/certifi-2023.7.22.tar.gz"
+    sha256 "539cc1d13202e33ca466e88b2807e29f4c13049d6d87031a3c110744495cb082"
+  end
```
Comment (chenrui333) [inline Formula/i/instaloader.rb:15]: we can vendor this out using `python-certifi` (also need to update pypi mapping file)

File: Formula/i/instaloader.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    url "https://files.pythonhosted.org/packages/36/dd/a6b232f449e1bc71802a5b7950dc3675d32c6dbc2a1bd6d71f065551adb6/urllib3-2.1.0.tar.gz"
+    sha256 "df7aa8afb0148fa78488e7899b2c59b5f4ffcfa82e6c54ccb9dd37c1d7b52d54"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "instaloader.github.io", shell_output("#{bin}/instaloader --help")
```
Comment (chenrui333) [inline Formula/i/instaloader.rb:42]: need some substantial test

File: Formula/i/instaloader.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    url "https://files.pythonhosted.org/packages/36/dd/a6b232f449e1bc71802a5b7950dc3675d32c6dbc2a1bd6d71f065551adb6/urllib3-2.1.0.tar.gz"
+    sha256 "df7aa8afb0148fa78488e7899b2c59b5f4ffcfa82e6c54ccb9dd37c1d7b52d54"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    outut = shell_output("#{bin}/instaloader --fast-update notexist 2>&1")
```
Comment (bevanjkay) [inline Formula/i/instaloader.rb:38]: @chenrui333 Unfortunately the user "notexist" does indeed, exist 😃

Comment (chenrui333) [reply Formula/i/instaloader.rb:38]: yeah, I know it does exist :) (but it is private page)

Comment (chenrui333) [reply Formula/i/instaloader.rb:38]: I actually see something different in my local

```
$ /opt/homebrew/Cellar/instaloader/4.10.1/bin/instaloader --fast-update notexist

HTTP redirect from https://i.instagram.com/api/v1/users/web_profile_info/?username=notexist to https://i.instagram.com/accounts/login/?next=/api/v1/users/web_profile_info/
notexist: Redirected to login page. Use --login.

Errors or warnings occurred:
notexist: Redirected to login page. Use --login.
```

Comment (chenrui333) [reply Formula/i/instaloader.rb:38]: actually the page got changed now 🤯 

image

Comment (chenrui333) [reply Formula/i/instaloader.rb:38]: might need a better account for the testing :(

```
    Minitest::Assertion: Expected /notexist:\ Redirected\ to\ login\ page\.\ Use\ \-\-login/ to match "Stored ID 24114340 for profile notexist.\nHint: Use --login to download higher-quality versions of pictures.\n[1/1] Downloading profile notexist\nnotexist/2015-08-12_06-04-16_UTC_profile_pic.jpg notexist: --login=USERNAME required.\n\nErrors or warnings occurred:\nnotexist: --login=USERNAME required.\n\n".
```

Comment (bevanjkay) [reply Formula/i/instaloader.rb:38]: What about the official instagram account?

**Comment by bevanjkay** [comment]
Tests seem to be a bit flaky, depending on network access. It is passing locally.

**Review by branchv — APPROVED** [review]

**Pull request merged**